packages/hub/utils/scheduled-payments.ts

```diff
@@ -1,30 +1,38 @@
-import { startOfDay } from 'date-fns';
+import { addMonths } from 'date-fns';
 import { convertDateToUTC } from './dates';
 
 // This function will return the next payment date based on the frequency.
 // fromDate will be either the current date when a recurring scheduled is created,
 // or the date of the last payment.
 // When a recurring scheduled is created the recurringUntil parameter must be not undefined.
-export function calculateNextPayAt(fromDate: Date, recurringDay: number, recurringUntil: Date) {
+// currentDate is needed for testing purpose.
+export function calculateNextPayAt(
+  recurringDay: number,
+  recurringUntil: Date,
+  options?: {
+    lastPayAt?: Date;
+    currentDate?: Date;
+  }
+) {
   if (recurringDay < 1 || recurringDay > 31) {
     throw new Error('Recurring day must be in the range of 1-31');
   }
+  let nowUtc = convertDateToUTC(options?.currentDate ?? new Date());
+  let recurringUntilUtc = convertDateToUTC(recurringUntil);
 
-  let nextPayAtUtc = convertDateToUTC(fromDate);
-
-  // We take min() because some months can have less days than the recurring day (e.g. there is no Feb 31)
-  let adjustedRecurringDay = Math.min(recurringDay, daysInMonth(fromDate));
-  if (fromDate.getUTCDate() < adjustedRecurringDay) {
-    nextPayAtUtc.setUTCDate(adjustedRecurringDay); // next pay this month
+  let nextPayAt;
+  if (options?.lastPayAt) {
+    nextPayAt = addMonths(options.lastPayAt, 1);
```

Comment:
Is this a bug?

Example:

1. recurring day is 31th
2. previous pay at is Feb 28
3. next pay at will be defined as Feb 28 + 1 month = March 28 (while it should be March 31)

Reply:
The real date will be defined here. https://github.com/cardstack/cardstack/pull/3728/files/6fa297acc40143b44d2bc89b3637afbf11dd5ca1#diff-f119d212bea4bffe5262e46a0bdc30bde8add6e018354e0ecb621269dd5a5ce7R32